pkg/infrastructure/gcp/clusterapi/firewallrules.go

```diff
@@ -198,6 +225,29 @@ func createFirewallRules(ctx context.Context, in clusterapi.InfraReadyInput, net
 		return err
 	}
 
+	// etcd are needed for master communication for etcd nodes
+	firewallName = fmt.Sprintf("%s-etcd", in.InfraID)
+	srcTags = []string{masterTag}
+	targetTags = []string{masterTag}
+	srcRanges = []string{}
+	if err := addFirewallRule(ctx, firewallName, network, projectID, getEtcdPorts(), srcTags, targetTags, srcRanges); err != nil {
+		return err
+	}
+
+	// Add a single firewall rule to allow the Google Cloud Engine health checks to access all of the services.
+	// This rule enables the ingress load balancers to determine the health status of their instances.
+	firewallName = fmt.Sprintf("%s-health-checks", in.InfraID)
+	srcTags = []string{}
+	targetTags = []string{masterTag}
+	srcRanges = []string{"35.191.0.0/16", "130.211.0.0/22"}
```

Comment:
Hm my first reaction when I see these is that they should go upstream. I suspect we added the initial firewall rules here out of expedience (not enough time to get them upstream), but I'm not certain of the background.

I don't think this should hold up the PR or anything, but we should probably consider adding these upstream.. perhaps a firewall rule API.